### PR TITLE
Add support for Defining DacPac deployment options in SqlDatabaseProjects.

### LINKED ISSUE #282: [Feature] Allow setting DacDeployOptions in DacpacDeployer when publishing SqlServer porjects from dacpac
**Background**
I'm working in a project where we have a base dacpac with common schema elements for multiple instances of different databases building on top of the common schema elements. The dacpac is referenced as a PackageReference in the SqlServer project and normally published with the `IncludeCompositeObjects=true` option to deploy alle schema elements required.

As we attempt to adopt .Net Aspire, we are not easily able to do this without re-implementing the functionality of the `CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects` package only to override the use of `new DacDeployOptions()` in `DacPacDeployer.Deploy()`.

It seems like a very usable feature to allow modifying the many `DacDeployOptions` properties passed to the Deploy operation for the dacpac in question.

**Proposal**
I thus propose an overload of the WithDacpac extension method for `SqlProjectResource `resource builders to set a new `DacDeployOptionsMetaDataAnnotation `which can be picked up when deploying the dacpac in the `SqlProjectPublishService` class.

Alternatively, the `DacDeployOptions` can be created and populated with supported property values from the sqlproj file in a new `GetDacDeployOptions` method in the `SqlProjectResource` class similar to how the `SqlTargetPath` and `TargetPath` are attempted read in the `GetDacpacPath` method.

**Help us help you**
Yes, I'd like to be assigned to work on this item, but am new to contributing on github.

### PR DESCRIPTION
**Pull request by nnitkasw**
This should allow specifying dacpac deployment options.

**Closes #282**

## PR Checklist

- [X] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [X] Based off latest main branch of toolkit
- [X] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [ ] New integration
  - [ ] Docs are written
  - [ ] Added description of major feature to project description for NuGet package (4000 total character limit, so don't push entire description over that)
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Contains **NO** breaking changes
- [X] Every new API (including internal ones) has full XML docs
- [X] Code follows all style conventions

## Other information

Public members added to the unshipped api documentation.
Includes tests for when options are used and not used.
The readme was updated with an eaxmple of the usage.

This my first contribution on github. I welcome feedback.

### CODE REVIEW

**Comment by nnitkasw** [comment]
@dotnet-policy-service agree

**Review by aaronpowell — COMMENTED** [review]
Syntactically, this looks all good but I'll defer to @jmezach as a SME on it

File: src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/PublicAPI.Unshipped.txt
Comment (jmezach) [inline src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/PublicAPI.Unshipped.txt:1]: @aaronpowell Just out of curiosity, but did the PublicAPI.Shipped.txt file get automagically updated when we released 9.0? I don't remember doing this, but it must have happened somewhere so I'm just curious how that works exactly ;)

File: src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/DacpacDeployer.cs
Comment (jmezach) [inline src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/DacpacDeployer.cs:16]: What's the reason for this change? I'm not necessarily against it, but it feels unrelated?

File: src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs
Comment (jmezach) [inline src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs:9]: I'm wondering if this needs to be `public`. It's basically just a mechanism for us to transport the delegate to the place we need it so I guess it could be `internal`. But maybe the Aspire convention is to have annotations always be public. Perhaps @aaronpowell could shed some light on this?

File: src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/DacpacDeployer.cs
Comment (nnitkasw) [reply src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/DacpacDeployer.cs:16]: Sorry, just a little boy scout mentality. It is not related.
I got code analysis warnings locally on the unused parameter and on the logging call. So I refactored according to the latest practices with source generated logger messages for parformance. Reference: https://learn.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging
But, in retrospect, the logging part is probably overkill given that this is never used in an running application, only for deployments. If you prefer, I can remove it to keep the code a little less complex and thus easier to maintain.

Comment (nnitkasw) [reply src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/DacpacDeployer.cs]: I have added new commit undoing the unrelated logging changes  except for the adding "using" of the disposable `DacPackage`.

File: src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs
Comment (aaronpowell) [reply src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs:9]: Having the record public simplifies testing (we don't have to rely on internals visible attributes). and it's consistent with annotations in the Aspire source, so let's leave it as `public`.

Comment (jmezach) [reply src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs:9]: 👍

**Review by jmezach — APPROVED** [review]
LGTM

**Comment by nnitkasw** [comment]
Not sure what the process is once the PR is approved.
But I'm closing the PR to complete it, and will then wait for a merge by an owner.

**Comment by aaronpowell** [comment]
> @aaronpowell I'm not sure what the process is once the PR is approved. But I'm closing the PR to complete it, and will then wait for a merge by your or another owner. Let me know if it needs to be reopened for it to be merged.

Needed to be open for merging 😉

Would you be able to update the [docs](https://learn.microsoft.com/en-gb/dotnet/aspire/community-toolkit/hosting-sql-database-projects?tabs=dotnet-cli) too @nnitkasw? The repo for that is here: https://github.com/dotnet/docs-aspire

**Pull request merged**
